types/google-apps-script/google-apps-script.spreadsheet.d.ts

```diff
@@ -1963,6 +2015,7 @@ declare namespace GoogleAppsScript {
             addDeveloperMetadata(key: string, value: string): Sheet;
             addDeveloperMetadata(key: string, value: string, visibility: DeveloperMetadataVisibility): Sheet;
             appendRow(rowContents: any[]): Sheet;
+            asDataSourceSheet(): DataSourceSheet;
```

Comment:
```suggestion
            asDataSourceSheet(): DataSourceSheet | null;
```

> Returns the sheet as a DataSourceSheet if the sheet is of type SheetType.DATASOURCE, or null otherwise. 
https://developers.google.com/apps-script/reference/spreadsheet/sheet#asDataSourceSheet()